src/rgw/rgw_crypt.cc

```diff
@@ -35,13 +35,13 @@ class canonical_char_sorter {
 private:
     const icu::Normalizer2* normalizer;
     CephContext *cct;
-
+    const DoutPrefixProvider *dpp;
 public:
-    canonical_char_sorter(CephContext *cct) : cct(cct) {
+    canonical_char_sorter(CephContext *cct, const DoutPrefixProvider *dpp) : cct(cct), dpp(dpp) {
```

Comment:
We decided last december to have the DoutPrefixProvider be the first argument to a function/constructor whenever a new one is being added. There are many places where this is not the case because they were added before we made this decision, but most of the dpp's should be the first argument to their function/constructor.